### ISSUE TITLE
Should GG distinguish between 'success' return code from a .exe file and when that .exe file is blocked by AV software from running?

### ISSUE THREAD

**Issue by G4OEU** [DistributedProofreaders/guiguts#1044]
**Describe the bug**
Not sure whether behaviour is bug or feature. Installed 1.5.0 with its jeebies.exe file. Norton AV does not (yet) trust that file so blocks it from running. However GG responds as if jeebies.exe has run and found no issues with file contents. The ErrorCheck window after running Tools>Jeebies says:

  Begining check: Jeebies
  Check is complete: Jeebies

**To Reproduce**
0. Need to have AV software installed that does not trust jeebies.exe and so blocks its execution
1. Open Guiguts with [jeebies-test.txt](https://github.com/DistributedProofreaders/guiguts/files/9901597/jeebies-test.txt)
2. Open the Tools menu ...
3. Scroll down to 'Run Jeebies'
4. See messages in ErrorCheck window

**Expected behavior**
If GG had, in fact, run jeebies.exe then the ErrorCheck window should have said for the test file above:

  Beginning check: Jeebies
    --> Odd file. There are 1 "he"s and no "be"s.
  1:55 - Query phrase "should he taken" 
  Check is complete: Jeebies

If GG has been unable to run jeebies.exe because AV software has blocked it then GG should say so and not give a misleading 'success' error status that suggests both that it ran jeebies.exe and the latter found nothing amiss with the file contents.

 Note that before GG responds as above, there is a popup from Norton AV saying that it 'blocked a suspicious action by jeebies.exe.'

**Environment**
* Guiguts version: 1.5.0 (official release)
* OS and version: Windows 10
* Perl version: 
* Perl distribution: 

**Screenshots**

**Additional context**
Running up to date Norton Anti-Virus software on desktop m/c.
Would expect similar 'incorrect' behaviour when GG is blocked from running any .exe file of external tools that are not 'trusted' by the AV software installed on a user's system?

**Comment by G4OEU** [comment]
Nigel,

Thanks for the reply. I will investigate further along the lines you
suggest and get back to you in a day or two.

Quentin

**Comment by G4OEU** [comment]
TL;DR The errors.err file is empty!

Using the modified ErrorCheck.pm I managed to recreate the situation where Norton AV 'blocked' an action when Tools>Jeebies was invoked.

1. A popup from Norton said "Suspicious Action Blocked".
2. The error check window from GG said "Beginning check: Jeebies\nCheck is complete: Jeebies".
3. The file ~\errors.err is present but empty.
4. The Norton activity logs says "Action observed: Suspicious process attempted writing into a file protected by Data Protector" and "Target: errors.err".
5. All the above implies that jeebies.exe was actually executed and that it attempted to write to the file errors.err in the same directory from which GG read the input test file used. That write to errors.err was blocked by Norton.

Are there any other circumstances in which the errors.err file would be empty after jeebies.exe was run? If not then could that be a test by GG to detect that jeebies.exe failed to complete as expected?

**Comment by G4OEU** [comment]
I accept that this is an AV software issue but it's a pity that GG cannot deal with it. However if users chose to protect their systems with AV s/w then they've got to expect a bit of inconvenience as the price for that protection. As for Norton's 'weird' behaviour I guess it cannot (easily?) detect that the file jeebies.exe wants to write to was created by the same program. I suppose it would be the simple solution for AV software to block all writes to files if the software that is doing the writing is not trusted. 

I can tell Norton to trust jeebies.exe although it was not obvious how to do this initially. I had an exchange with Charlie about it and he provided some helpful tips. The issue, it turned out, was that I install GG, etc., and do all my work with it under a local, non-privileged, login ID. The Windows 10 system and Norton AV were installed using the privileged Microsoft login ID. Some features of Norton, including the ability to 'trust' an executable, seem only available from the login ID under which Norton was installed. Once I sussed that I can switch to my Microsoft login to fix problems with untrusted software. Note that I'm surprised that Norton allowed jeebies.exe to execute but I suppose it can't do much system damage when run under a non-privileged ID. However I would expect Norton to not allow untrusted software to execute at all under a privileged ID as that would allow all sorts of system exploits to be explored; I haven't verified that is the case yet but will use GG/jeebies.exe to test that.

Thanks for looking into this. There is an appropriate work around so I will close this issue.

**Comment by windymilla** [comment]
Found an appropriate way to trap this: #1046
